File: Core/Entities/Dtos/LogDto.cs
using System;

namespace Core.Entities.Dtos
{
    public class LogDto : IEntity
    {
        public int Id { get; set; }
        public string Level { get; set; }
        public string ExceptionMessage { get; set; }
        public DateTime TimeStamp { get; set; }
        public string User { get; set; }
        public string Value { get; set; }
        public string Type { get; set; }
    }
}

File: UiPreparation/FlutterUi/flutter_devarchitecture/lib/core/features/admin_panel/log_f/models/log_dto.dart
import '/core/models/i_dto.dart';

class LogDto implements IDto {
  int id;

  String level;

  String exceptionMessage;

  DateTime timeStamp;

  String user;

  String value;

  String type;

  LogDto({
    required this.id,
    required this.level,
    required this.exceptionMessage,
    required this.timeStamp,
    required this.user,
    required this.value,
    required this.type,
  });

  @override
  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'level': level,
      'exceptionMessage': exceptionMessage,
      'timeStamp': timeStamp,
      'user': user,
      'value': value,
      'type': type,
    };
  }

  factory LogDto.fromMap(Map<String, dynamic> map) {
    return LogDto(
      id: map['id'],
      level: map['level'] ?? "",
      exceptionMessage: map['exceptionMessage'] ?? "",
      timeStamp: DateTime.parse(map['timeStamp'] ?? DateTime.now()),
      user: map['user'] ?? "",
      value: map['value'] ?? "",
      type: map['type'] ?? "",
    );
  }
}
